routers/web/base.go

```diff
@@ -44,7 +45,7 @@ func storageHandler(storageSetting setting.Storage, prefix string, objStore stor
 				routing.UpdateFuncInfo(req.Context(), funcInfo)
 
 				rPath := strings.TrimPrefix(req.URL.Path, "/"+prefix+"/")
-				rPath = path.Clean("/" + strings.ReplaceAll(rPath, "\\", "/"))[1:]
+				rPath = util.CleanPath(strings.ReplaceAll(rPath, "\\", "/"))
```

Comment:
This is wrong and is going to result in it being possible to have rPath have a preceding "/" whereas previously it was impossible.

Reply:
If rPath has a `/` prefix, it will be invoked with `path.Clean` directly. I think it should be better than the current logic.

Reply:
Even if `util.CleanPath` here will not convert absolute path to relative path, but line 47 will ensure rPath is a relative path.

Reply:
I think we need clear definitions for every case, instead of using unpredictable `CleanPath` behavior.

The unstable part is :

* `CleanPath("/path")` => `/path`
* `CleanPath("path")` => `path`

But in many cases, we only want `path`.

Reply:
But that's not `CleanPath`'s problem. I sent #23446 to fix possible unclear places

Reply:
I think the problem is that `CleanPath`'s behavior is not that stable in many cases.

Callers should always know what they need - absolute or relative - no matter what path has been passed in the util function.

Reply:
absolute or relative should not be the responsibility of `CleanPath`. It should be decided out of the function. Or we can rename `CleanPath` to `CleanAndEnsureRelativePath`.

Reply:
Hmm .. that's a disagreement: I do not think so about "absolute or relative should not be the responsibility of CleanPath. It should be decided out of the function."

IMO maybe it could be (like #23441)

```go
rel := util.PathRelJoin(....)   //  `/path` => `path`,  `path` => `path`
abs := path.Join("/", util.PathRelJoin(....))  // if necessary
```